doc/sql.txt

```diff
@@ -0,0 +1,436 @@
+
+
+    Parameters: ~
+        {tbl}   (string)  the name of the db table.
+        {specs} (table)   a {spec} or a list of {specs} with where and values
+                          key.
+
+    Return: ~
+        boolean: true if operation is successfully, false otherwise.
+
+    Usage: ~
+        db:delete("todos")
+        db:delete("todos", { where = { id = 1 })
+        db:delete("todos", { where = { id = {1,2,3} })
+
+
+sql:select()                                                    *sql:select()*
+    Query from a table with where and join options
+
+
+    Return: ~
+        lua list of matching rows
+
+    Usage: ~
+        db:get("todos") -- everything
+        db:get("todos", { where = { id = 1 })
+        db:get("todos", { limit = 5 })
+
+
+sql:table({tbl_name})                                            *sql:table()*
+    Create new sql-table object.
+
+
+    Parameters: ~
+        {tbl_name} (string)  the name of the table. can be new or existing
+                             one.
+
+    Return: ~
+        table
+
+
+
+================================================================================
+tbl:schema({schema})                                            *tbl:schema()*
+    Create or change {self.name} schema. If no {schema} is given, then it
+    return current the used schema.
+
+
+    Parameters: ~
+        {schema} (table)  table schema definition
+
+    Return: ~
+        table: list of keys or keys and their type.
+
+
+tbl:drop()                                                        *tbl:drop()*
+    Same functionalities as |sql:drop()|, if the table is already drooped then
+    it returns false
+
+
+    Return: ~
+        boolean
+
+
+tbl:empty()                                                      *tbl:empty()*
+    Predicate that returns true if the table is empty
+
+
+    Return: ~
+        boolean
+
+
+tbl:exists()                                                    *tbl:exists()*
+    Predicate that returns true if the table exists
+
+
+    Return: ~
+        boolean
+
+
+tbl:count()                                                      *tbl:count()*
+    The count of the rows in {self.name}.
+
+
+    Return: ~
+        number: number of rows in {self.name}
+
+
+tbl:get({query})                                                   *tbl:get()*
+    Query the table and return results. If the {query} has been ran before,
+    then query results from cache will be returned.
+
+
+    Parameters: ~
+        {query} (table)  query.where, query.keys, query.join
+
+    Return: ~
+        table: empty table if no result
+
+    See: ~
+        |sql:select()|
+
+
+tbl:where({where})                                               *tbl:where()*
+    Get first match from a table based on keys query results from cache will be
+    returned.
+
+
+    Parameters: ~
+        {where} (table)  where key values
+
+    Return: ~
+        nil or row
+
+    Usage: ~
+        ---@usage: tbl:where{id = 1}
```

Comment:
maybe bug

Reply:
maybe maybe not. it just doesn't like `:` i remove it ;)